core/core.php

```diff
@@ -181,4 +181,86 @@ public function validate_badge_order($user_id)
 
 		return true;
 	}
+
+	/**
+	 * Gets the url to the user avatar.
+	 * Returns empty string on failure or if there is no avatar.
+	 *
+	 * @param array $user_data Avatar data from the user table
+	 * @return array
+	 */
+	public function get_avatar_data($user_data)
+	{
+		global $user, $config;
+		global $phpbb_container, $phpbb_dispatcher;
+
+		if (!$config['allow_avatar'])
+		{
+			return false;
+		}
+
+		$row = [
+			'avatar' => $user_data['user_avatar'],
+			'avatar_type' => $user_data['user_avatar_type'],
+			'avatar_width' => $user_data['user_avatar_width'],
+			'avatar_height' => $user_data['user_avatar_height'],
+		];
+
+		$avatar_data = [
+			'src' => $row['avatar'],
+			'width' => $row['avatar_width'],
+			'height' => $row['avatar_height'],
+		];
+
+		/* @var $phpbb_avatar_manager \phpbb\avatar\manager */
+		$phpbb_avatar_manager = $phpbb_container->get('avatar.manager');
+		$driver = $phpbb_avatar_manager->get_driver($row['avatar_type']);
+
+		if (!$driver)
+		{
+			return false;
+		}
+
+		$avatar_data = $driver->get_data($row);
+
+		if ($row['avatar_type'] === 'avatar.driver.gravatar')
+		{
+			$avatar_data['src'] = $this->get_gravatar_url($row);
+		}
+
+		return $avatar_data;
+	}
+
+	/**
+	 * Gets the URL for a gravatar.
+	 * Essentially a copy of the protected method form avatar.driver.gravatar
+	 * 
+	 * @param $row User data
+	 * @return string Gravatar URL
+	 */
+	protected function get_gravatar_url($row)
+	{
+		global $phpbb_dispatcher;
+
+		$url = '//secure.gravatar.com/avatar/';
```

Comment:
Starting a url with just `//` might be valid for a web browser (TIL), but its not for most other things. Can we please specify `https://` here?